### PR TITLE
Coref data fix

### PR DESCRIPTION
**Pull request by DeNeutoy**
This reverts part of #584 which switched from using v4 Ontonotes annotations for the 2012 Conll coref task to using labels from v5, which was incorrect and contained instances which made the model OOM. This fixes that and reads exactly the same data as previously, but utilising the new `Ontonotes` reader rather than the original one.

### CODE REVIEW

File: scripts/compile_coref_data.sh [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+download_and_extract $conll_url conll-2012-scripts.v3.tar.gz
+
+download_and_extract http://conll.cemantix.org/download reference-coreference-scorers.v8.01.tar.gz
+mv reference-coreference-scorers conll-2012/scorer
+
+# Convert the ontonotes data into the CONLL format.
+bash conll-2012/v3/scripts/skeleton2conll.sh -D $ONTONOTES_PATH/data/files/data conll-2012
+
+compile_language english
```
Comment (matt-gardner) [inline scripts/compile_coref_data.sh]: Can you add a newline here?

File: tests/fixtures/coref/coref.gold_conll [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+bc/cctv/00/cctv_0000   3   15    developing   VBG         (S(VP*    develop  02   -   Speaker#1       *             *      *         *      (V*)     -
+bc/cctv/00/cctv_0000   3   16           the    DT           (NP*         -    -   -   Speaker#1       *             *      *         *   (ARG1*      -
+bc/cctv/00/cctv_0000   3   17          Hong   NNP          (NML*         -    -   -   Speaker#1   (GPE*             *      *         *        *     (6
+bc/cctv/00/cctv_0000   3   18          Kong   NNP              *)        -    -   -   Speaker#1       *)            *      *         *        *      6)
+bc/cctv/00/cctv_0000   3   19       tourism    NN              *         -    -   -   Speaker#1       *             *      *         *        *      -
+bc/cctv/00/cctv_0000   3   20      industry    NN         *))))))        -    -   -   Speaker#1       *             *)     *         *        *)     -
+bc/cctv/00/cctv_0000   3   21             .     .             *))        -    -   -   Speaker#1       *             *      *         *        *      -
+ 
+#end document
```
Comment (matt-gardner) [inline tests/fixtures/coref/coref.gold_conll]: A newline here would also be nice.

File: tests/fixtures/coref/experiment.json [hunk 3]
```diff
@@ -15,8 +15,8 @@
     },
     "max_span_width": 5
   },
-  "train_data_path": "tests/fixtures/conll_2012/",
```
Comment (matt-gardner) [inline tests/fixtures/coref/experiment.json]: Do we still need these old fixtures?  Are they used by the SRL model, or can they be removed?

Comment (DeNeutoy) [reply tests/fixtures/coref/experiment.json]: Unfortunately they are used by the SRL model, because that assumes only one document per file.

**Comment by DeNeutoy** [comment]
@roys174 This might effect you. The reader which is currently in master reads in coref examples correctly from the conll format (so if you have a similar reader for the Genia data you can keep using it), but I'd switched over to using a cleaner reader for the conll 2012 coref data. However, the Ontonotes 5.0 dataset is good for SRL but has incomplete coref annotations, which I didn't realise, which is why that processing script you asked about previously was required - it merged the _text_ from Ontonotes 5.0 with the coref labels from Ontonotes 4.0. I can come and chat about this change if you want to know more.

**Comment by tingkai-zhang** [comment]
Hi, @DeNeutoy, a following up question here.

I didn't successfully get the result when running this script.

To reproduce:

`bash compile_coref_data.sh data/`

It gave me, at the end of output:
```
----------------------------------------------------------------------------------------------------

could not find the gold parse [one//data/files/data/arabic/annotations/nw/ann/00/ann_0082.parse] in the ontonotes distribution ... exiting ...

----------------------------------------------------------------------------------------------------
cat: 'conll-2012/v4/data/development/data/english/annotations/*/*/*/*.v4_gold_conll': No such file or directory
cat: 'conll-2012/v4/data/train/data/english/annotations/*/*/*/*.v4_gold_conll': No such file or directory
cat: 'conll-2012/v4/data/test/data/english/annotations/*/*/*/*.v4_gold_conll': No such file or directory

```

Could you help me out on this? 

Thank you!

**Comment by djshowtime** [comment]
> Hi, @DeNeutoy, a following up question here.
> 
> I didn't successfully get the result when running this script.
> 
> To reproduce:
> 
> `bash compile_coref_data.sh data/`
> 
> It gave me, at the end of output:
> 
> ```
> ----------------------------------------------------------------------------------------------------
> 
> could not find the gold parse [one//data/files/data/arabic/annotations/nw/ann/00/ann_0082.parse] in the ontonotes distribution ... exiting ...
> 
> ----------------------------------------------------------------------------------------------------
> cat: 'conll-2012/v4/data/development/data/english/annotations/*/*/*/*.v4_gold_conll': No such file or directory
> cat: 'conll-2012/v4/data/train/data/english/annotations/*/*/*/*.v4_gold_conll': No such file or directory
> cat: 'conll-2012/v4/data/test/data/english/annotations/*/*/*/*.v4_gold_conll': No such file or directory
> ```
> Could you help me out on this?
> 
> Thank you!

I have the same problem as you.

**Comment by k1c** [comment]
Hi - is there any update on the error with the shell script? I'm running into the same issue.

**Pull request merged**